src/test/java/com/connexta/ingest/service/impl/IngestServiceImplUnitTests.java

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.ingest.service.impl;
+
+import static com.connexta.ingest.service.impl.IngestServiceImpl.METACARD_MEDIA_TYPE;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.connexta.ingest.adaptors.MetacardStorageAdaptor;
+import com.connexta.ingest.client.StoreClient;
+import com.connexta.ingest.client.TransformClient;
+import com.connexta.ingest.exceptions.StoreException;
+import com.connexta.ingest.exceptions.StoreMetacardException;
+import com.connexta.ingest.exceptions.TransformException;
+import com.connexta.ingest.service.api.IngestService;
+import java.io.InputStream;
+import java.net.URI;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class IngestServiceImplUnitTests {
+
+  @Mock private StoreClient mockStoreClient;
+  @Mock private MetacardStorageAdaptor mockMetacardStorageAdaptor;
+  @Mock private TransformClient mockTransformClient;
+
+  // TODO test invalid constructor arguments
+
+  @Test
+  public void testIngest() throws Exception {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "http://localhost:9040/ingest/",
+            mockTransformClient);
+
+    final Long fileSize = 1L;
+    final String mimeType = "mimeType";
+    final InputStream inputStream = mock(InputStream.class);
+    final String fileName = "fileName";
+    final long metacardFileSize = 1L;
+    final InputStream metacardInputStream = mock(InputStream.class);
+    final URI location = new URI("http://localhost:9041/mis/product/1234");
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName)).thenReturn(location);
+
+    // when
+    assertDoesNotThrow(
+        () ->
+            ingestService.ingest(
+                fileSize, mimeType, inputStream, fileName, metacardFileSize, metacardInputStream));
+
+    // then
+    final InOrder inOrder =
+        Mockito.inOrder(mockStoreClient, mockMetacardStorageAdaptor, mockTransformClient);
```

Comment:
`mockStoreClient` is not needed as an argument unless there is a `inorder.verify(mockStoreClient)` statement.

Reply:
The verify statements are right below.